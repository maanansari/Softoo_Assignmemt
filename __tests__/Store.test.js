import Store from '../src/screens/StoreScreen';
import {fireEvent, render} from '@testing-library/react-native';
import {Provider} from 'react-redux';
import {configureStore} from '@reduxjs/toolkit';
import rootReducer from '../src/redux/rootReducer';

describe('render store', () => {
  let store = configureStore({
    reducer: rootReducer,
  });

  let navigation = {
    goBack: jest.fn(),
  };
  let component = (
    <Provider store={store}>
      <Store navigation={navigation} />
    </Provider>
  );

  it('match snap', () => {
    const tree = render(component);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('render children', async () => {
    const tree = render(component);
    const backBtn = await tree.findByTestId('back-btn');
    fireEvent.press(backBtn);
    expect(navigation.goBack).toBeCalled();
    expect(backBtn).toBeDefined();
  });

  it('render back icon', async () => {
    const tree = render(component);
    const backBtn = await tree.findByTestId('go-icon');
    expect(backBtn.props.source.testUri).toBe('../../../src/assets/goback.png');
  });

  it('render flatList', async () => {
    const tree = render(component);
    const flatList = await tree.findByTestId('store-list');
  });
});
