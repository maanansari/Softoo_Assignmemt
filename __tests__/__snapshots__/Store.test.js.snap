// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render store match snap 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "height": 35,
        "marginStart": 8,
        "marginTop": 2,
      }
    }
    testID="back-btn"
  >
    <Image
      source={
        {
          "testUri": "../../../src/assets/goback.png",
        }
      }
      style={
        {
          "height": "100%",
          "width": "10%",
        }
      }
      testID="go-icon"
    />
  </View>
  <RCTScrollView
    ListEmptyComponent={[Function]}
    contentContainerStyle={
      {
        "paddingVertical": 20,
      }
    }
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    testID="store-list"
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            {
              "color": "#000",
              "fontSize": 14,
            }
          }
        >
          No Item in Cart
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
